python/rikai/spark/sql/codegen/pytorch.py

```diff
@@ -41,20 +41,23 @@ def generate_udf(spec: "rikai.spark.sql.codegen.base.ModelSpec"):
     -------
     A Spark Pandas UDF.
     """
-
-    options = {} if spec.options is None else spec.options
-    use_gpu = options.get("device", "cpu") == "gpu"
+    use_gpu = spec.options.get("device", "cpu") == "gpu"
     num_workers = int(
-        options.get("num_workers", min(os.cpu_count(), DEFAULT_NUM_WORKERS))
+        spec.options.get(
+            "num_workers", min(os.cpu_count(), DEFAULT_NUM_WORKERS)
+        )
     )
-    batch_size = int(options.get("batch_size", DEFAULT_BATCH_SIZE))
+    batch_size = int(spec.options.get("batch_size", DEFAULT_BATCH_SIZE))
 
     def torch_inference_udf(
         iter: Iterator[pd.DataFrame],
     ) -> Iterator[pd.DataFrame]:
 
-        with open_uri(spec.uri) as fobj:
-            model = torch.load(fobj)
+        if spec.artifact:
+            model = spec.artifact
```

Comment:
we should consolidate these two methods.

Reply:
the mlflow one isn't just a simple file. it's Mlflow Model format with pickle information as well.

Reply:
yea, what i meant is that we can deligate the "load model" impl behind "ModelSpec"

For example: `FileSystemModelSpec.load_model()` and `MLflowModelSpec.load_model()`, so that we dont need `if/else` here.